airbyte-webapp/src/core/form/schemaToYup.ts

```diff
@@ -0,0 +1,208 @@
+    // if a key doesn't have a sub schema for a selection const value, set it to "strip"
+    // so it's removed before the form is sent to the server
+    return yup.object().shape(
+      Object.fromEntries(
+        Array.from(flattenedKeys.entries()).map(([key, schemaByCondition]) => {
+          let mergedSchema = yup.mixed();
+          if (key === selectionKey) {
+            // do not validate the selectionKey itself, as the user can't change it it doesnt matter
+            return [key, mergedSchema];
+          }
+          const allSelectionConstValuesWithThisKey = schemaByCondition.map(([constValue]) => constValue);
+          schemaByCondition.forEach(([selectionConstValue, conditionalSchema]) => {
+            mergedSchema = mergedSchema.when(selectionKey, {
+              is: selectionConstValue,
+              then: () => conditionalSchema,
+              otherwise: (schema) => schema,
+            });
+          });
+          mergedSchema = mergedSchema.when(selectionKey, {
+            is: (val: unknown) => !allSelectionConstValuesWithThisKey.includes(val),
+            then: (schema) => schema.strip(),
+            otherwise: (schema) => schema,
+          });
+          return [key, mergedSchema];
+        })
+      )
+    );
+  }
+
+  switch (jsonSchema.type) {
+    case "string":
+      schema = yup
+        .string()
+        .transform((val) => String(val))
+        .trim();
+
+      if (jsonSchema?.pattern !== undefined) {
+        schema = schema.matches(new RegExp(jsonSchema.pattern), "form.pattern.error");
+      }
+
+      break;
+    case "boolean":
+      schema = yup.boolean();
+      break;
+    case "integer":
+    case "number":
+      schema = yup.number().transform((value) => (isNaN(value) ? undefined : value));
+
+      if (jsonSchema?.minimum !== undefined) {
+        schema = schema.min(jsonSchema?.minimum);
+      }
+
+      if (jsonSchema?.maximum !== undefined) {
+        schema = schema.max(jsonSchema?.maximum);
+      }
+      break;
+    case "array":
+      if (typeof jsonSchema.items === "object" && !Array.isArray(jsonSchema.items)) {
+        schema = yup
+          .array()
+          .of(
+            buildYupFormForJsonSchema(
+              jsonSchema.items,
+              (formField as FormObjectArrayItem).properties,
+              jsonSchema,
+              propertyKey,
+              propertyPath ? `${propertyPath}.${propertyKey}` : propertyKey
+            )
+          );
+      }
+      break;
+    case "object":
+      let objectSchema = yup.object();
+
+      const keyEntries = Object.entries(jsonSchema.properties || {}).map(([propertyKey, propertySchema]) => {
+        const correspondingFormField = (formField as FormGroupItem).properties.find(
+          (property) => property.fieldKey === propertyKey
+        );
+        if (!correspondingFormField) {
+          throw new FormBuildError("mistmatch between form fields and schema");
```

Comment:
This seems like more of an internal bug, so we should probably track this error in datadog so we can be aware of it when it happens. We have a service (app monitoring service) which can be used to track errors like this. Here is an example: https://github.com/airbytehq/airbyte/blob/2392acb845891f9c031b06feee9af2a2f975f1a7/airbyte-webapp/src/views/Connector/ConnectorForm/useAuthentication.tsx#L156-L159
Here is a slack post with more context about this service: https://airbytehq-team.slack.com/archives/C039SG3DQ8K/p1668506492916659

In fact, we may just want to track all FormBuildErrors?

Reply:
Good idea, added that to the error boundary - if a form group error bubbles up, it is tracking the error using datadog.